src/common/database.c

```diff
@@ -967,6 +967,46 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
 
     sqlite3_exec(db->handle, "COMMIT", NULL, NULL, NULL);
     new_version = 15;
+  }
+  else if(version == 15)
+  {
+    sqlite3_stmt *get_images_stmt = NULL;
+    sqlite3_stmt *update_stmt = NULL;
+    int max_position = 0;
+
+    sqlite3_exec(db->handle, "BEGIN TRANSACTION", NULL, NULL, NULL);
+    ////////////////////////////// custom image order
+    TRY_EXEC("ALTER TABLE main.images ADD COLUMN position INTEGER",
+             "[init] can't add `position' column to images table in database\n");
+    TRY_EXEC("CREATE INDEX main.image_position_index ON images (position)",
+             "[init] can't create index for custom image order table\n");
+
+    // get images to add a position
+    gchar * image_query = g_strdup("SELECT id FROM main.images ORDER BY filename");
+    DT_DEBUG_SQLITE3_PREPARE_V2(db->handle, image_query, -1, &get_images_stmt, NULL);
+
+    // update position for existing images
+    gchar *update_query = g_strdup("UPDATE main.images SET position = ?1 WHERE id = ?2");
+    DT_DEBUG_SQLITE3_PREPARE_V2(db->handle, update_query, -1, &update_stmt, NULL);
+
+    while(sqlite3_step(get_images_stmt) == SQLITE_ROW)
```

Comment:
Likewise this could take even longer to run for the user with 100K+ images?

Reply:
I'll think about this too.